project1/server/server_tasks/server_task.cpp

```diff
@@ -0,0 +1,65 @@
+#include "server_task.h"
+namespace server_tasks {
+
+    thread_pool::Task::Status ServerTask::SetUp()  {
+        // Create the socket.
+        const auto kAddress = MakeAddress(port_);
+
+        server_fd_ = SetUpSocket(kAddress);
+        if (server_fd_ < 0) {
+            return thread_pool::Task::Status::FAILED;
+        } else return thread_pool::Task::Status::RUNNING;
+    }
+
+    thread_pool::Task::Status ServerTask::RunAtomic() {
+
+        return Listen();
+
+    }
+
+    void ServerTask::SetPort(uint16_t port) {
+        port_ = port;
+    }
+
+    void ServerTask::SetActiveCommandIDs(std::shared_ptr<CommandIDs> cmd_ids) {
+        active_ids_ = cmd_ids;
+    }
+    struct sockaddr_in ServerTask::MakeAddress(uint16_t port) {
+        struct sockaddr_in address{};
+        address.sin_family = AF_INET;
+        address.sin_addr.s_addr = INADDR_ANY;
+        address.sin_port = htons(port);
+
+        return address;
+    }
+
+    int ServerTask::SetUpSocket(const struct sockaddr_in &address) {
+        // Open a TCP socket.
+        const int server_fd = socket(AF_INET, SOCK_STREAM, 0);
+        if (server_fd == 0) {
+            perror("Failed to create server socket");
```

Comment:
I know you copy/pasted this from my code, but let's still make these log messages instead of `perror`s.